### PR TITLE
Write the Basix hash into generated code (or 0 if the element is not plain C++ Basix element)

### PR DESCRIPTION
**Pull request by mscroggs**
Required for https://github.com/FEniCS/dolfinx/issues/3169.

DOLFINx PR that follows on from this: https://github.com/FEniCS/dolfinx/pull/3186

### CODE REVIEW

File: ffcx/codegeneration/ufcx.h [hunk 1]
```diff
@@ -162,7 +162,7 @@ extern "C"
     bool needs_facet_permutations;
 
     /// Get the coordinate element associated with the geometry of the mesh.
-    long int coordinate_element;
+    unsigned long int coordinate_element;
```
Comment (mscroggs) [inline ffcx/codegeneration/ufcx.h:165]: I'm getting a lot of `warning: integer constant is so large that it is unsigned` when compiling the generated code for this, but adjusting the type doesn't seem to make any difference

Comment (mscroggs) [reply ffcx/codegeneration/ufcx.h]: Adding `u` to the end of each literal int removed this warning

**Review by garth-wells — COMMENTED** [review]
Could we add `_hash` to the variable names in `ufcx.h`?

**Review by garth-wells — APPROVED** [review]
Is `u` the right suffix? 

Looks like it's all rather messy in C, https://en.cppreference.com/w/c/language/integer_constant.

**Pull request merged**